packages/flutter/lib/src/painting/binding.dart

```diff
@@ -62,8 +61,8 @@ mixin PaintingBinding on BindingBase, ServicesBinding {
   ///
   /// The image cache is created during startup by the [createImageCache]
   /// method.
-  ImageCache get imageCache => _imageCache;
-  ImageCache _imageCache;
+  ImageCache? get imageCache => _imageCache;
```

Comment:
When is the imageCache ever expected to be null?

/cc @dnfield

Reply:
No but it gets created by a virtual method. Can we make it late?

Reply:
I made it nullable because L117 there's `imageCache?.clear();`. I will change it to late non-nullable

Reply:
This change makes the following test fails:

```dart
  test('PaintingBinding with memory pressure before initInstances', () {
    // Observed in devicelab: the device sends a memory pressure event to us
    // after PaintingBinding has been created but before initInstances called,
    // meaning the imageCache member is still null.
    final PaintingBinding binding = TestPaintingBinding();
    expect(binding.imageCache, null);
    binding.handleMemoryPressure();
    expect(binding.imageCache, null);
    binding.initInstances();
    expect(binding.imageCache, isNotNull);
    expect(binding.imageCache.currentSize, 0);
  });
```

I guess I can remove it?

Reply:
Ahh I remember now.  This caused flakes sometimes because the memory pressure would get processed before finishing binding initialization on rare occasions.  I guess we have to leave it as nullable.

Reply:
Please don't remove this test, this was a real failure. @dnfield I think re-ordered some of the initialization but I don't think it is solved.

Specifically, we can have a queued up message for us when initializing the binding, which might be a low memory event. I believe that gets forwarded immediately, but due to the binding order will run before the image cache is initialized. That doesn't mean that the image cache needs to be nullable, since this state isn't generally observable - but you do need to track whether it has been initialized somewhere.

Reply:
Something like:
```dart
late ImageCache imageCache;
bool _imageCacheSetup = false;

```

Reply:
ok, after reading more carefully the comment I understood that it was not a good idea to remove it :)

Reply:
@jonahwilliams I'm not sure to follow. Do you expect a change in this PR or are you just talking about a future change that could happen?

Reply:
nullable is fine for now